proto/streamlit/proto/GitInfo.proto

```diff
@@ -0,0 +1,28 @@
+/**
+* Copyright 2018-2020 Streamlit Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+syntax = "proto3";
+
+// Message used to update page metadata.
+message GitInfo {
+  string repository = 1;
+  string branch = 2;
+  string module = 3;
+  bool is_head_detached = 4;
```

Comment:
Are `is_head_detached` and `is_ahead` exclusive (only one or the other is true)? If so, instead of having multiple booleans, consider having an enum state for it, e.g. 

```
  enum GitStates {
    DEFAULT = 0;
    HEAD_DETACHED = 1;
    AHEAD_OF_REMOTE = 2;
  }
  GitState state = 8;
```
This would have the advantages of:
- Clarifying that only one or the other is true
- Being easier to extend in the future, if there are other git repo states we care about

One general rule at Google about designing Protobuf APIs is to prefer enums over booleans, even if they only have two types! This is because a lot of the time you may end wanting to increase the number of options beyond just true/false, and an enum scales more gracefully. See also: https://thoughtbot.com/blog/booleans-and-enums